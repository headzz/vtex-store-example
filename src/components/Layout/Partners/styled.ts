import { GREY_LVL1 } from 'configuration/constants'
import styled from 'styled-components'

export const PartnerStyled = styled.section`
  background-color: ${GREY_LVL1};
  padding: 2.4rem 0;
  width: 100%;
`
