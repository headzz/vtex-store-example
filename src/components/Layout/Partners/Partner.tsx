import { PartnerStyled } from './styled'

/**
 * @component
 * @name Partner
 * @description Defines the container for the layout of Partner.
 * @namespace components components/Layout/Partner
 * @param {element} children Component(s) that may appear inside the layout.
 *
 */

const Partner: React.FC = ({ children }) => {
  return <PartnerStyled>{children}</PartnerStyled>
}

export default Partner
