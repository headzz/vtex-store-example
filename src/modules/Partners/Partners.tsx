import Layout from 'components/Layout'
import PartnersImage from './components/PartnersImage'
import TextContainer from './components/TextContainer'
import { PartnerContainer } from './styled'

/**
 * @component
 * @name Partners
 * @description creates Partners component section to use on screen.
 * @namespace components modules/Partners
 *
 */
const Partners: React.FC = () => {
  return (
    <Layout.Partner>
      <PartnerContainer>
        <TextContainer />
        <PartnersImage />
      </PartnerContainer>
    </Layout.Partner>
  )
}

export default Partners
