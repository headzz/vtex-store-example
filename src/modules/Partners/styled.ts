import styled from 'styled-components'

export const PartnerContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  flex-direction: column;

  p:nth-child(2) {
    display: none;
  }

  & > div {
    margin-bottom: 1.2rem;
  }

  p:first-child {
    margin-bottom: 1.2rem;
  }

  @media (min-width: 768px) {
    flex-direction: row;

    & > div {
      margin-bottom: 0;
    }

    p:first-child {
      margin-bottom: 0;
    }

    p:nth-child(2) {
      display: block;
    }

    #arrow-right {
      display: block;
    }

    #arrow-down {
      display: none;
    }
  }
`
