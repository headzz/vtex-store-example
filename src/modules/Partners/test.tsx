import { render } from '@testing-library/react'

import Partners from './Partners'

describe('Partners', () => {
  it('match snapshot', () => {
    const { container } = render(<Partners />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
