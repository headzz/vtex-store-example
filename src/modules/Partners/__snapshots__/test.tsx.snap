// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Partners match snapshot 1`] = `
.c0 {
  background-color: #F2F2F2;
  padding: 2.4rem 0;
  width: 100%;
}

.c2 {
  font-size: 1.6rem;
  font-weight: 500;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 p:nth-child(2) {
  display: none;
}

.c1 > div {
  margin-bottom: 1.2rem;
}

.c1 p:first-child {
  margin-bottom: 1.2rem;
}

@media (min-width:768px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }

  .c1 > div {
    margin-bottom: 0;
  }

  .c1 p:first-child {
    margin-bottom: 0;
  }

  .c1 p:nth-child(2) {
    display: block;
  }

  .c1 #arrow-right {
    display: block;
  }

  .c1 #arrow-down {
    display: none;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <p
      class="c2"
      color=""
      title=""
    >
      Nossas principais lojas VTEX
    </p>
    <p
      class="c2"
      color=""
      title=""
    >
      â†’
    </p>
    <div>
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYwIiBoZWlnaHQ9IjI2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <img
          alt="logo da empresa brastemp"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
    <div>
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYwIiBoZWlnaHQ9IjI2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <img
          alt="logo da empresa compra certa"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
    <div>
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYwIiBoZWlnaHQ9IjI2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <img
          alt="logo da empresa consul"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
    <div>
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYwIiBoZWlnaHQ9IjI2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <img
          alt="logo da empresa the bar"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
  </div>
</section>
`
